recipes/aws-c-http/all/conanfile.py

```diff
@@ -66,15 +66,17 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, "lib", "aws-c-http"))
 
     def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "aws-c-http")
+        self.cpp_info.set_property("cmake_target_name", "AWS::aws-c-http")
+
         self.cpp_info.filenames["cmake_find_package"] = "aws-c-http"
         self.cpp_info.filenames["cmake_find_package_multi"] = "aws-c-http"
-        self.cpp_info.set_property("cmake_file_name", "aws-c-http")
         self.cpp_info.names["cmake_find_package"] = "AWS"
         self.cpp_info.names["cmake_find_package_multi"] = "AWS"
-        self.cpp_info.set_property("cmake_target_name", "AWS")
+
+        self.cpp_info.components["aws-c-http-lib"].set_property("cmake_target_name", "AWS::aws-c-http")
```

Comment:
I have set the cmake_target_name to the component as well, after this discussion: https://github.com/conan-io/conan-center-index/pull/8428/files#r770091435